pointgrey_camera_driver/src/nodelet.cpp

```diff
@@ -367,17 +367,62 @@ class PointGreyCameraNodelet: public nodelet::Nodelet
       }
       catch(std::runtime_error& e)
       {
+        // Something terrible has happened, so let's just disconnect and reconnect to see if we can recover.
         NODELET_ERROR("%s", e.what());
-        ///< @todo Look into readding this
-        /*try{
-          // Something terrible has happened, so let's just disconnect and reconnect to see if we can recover.
-          pg_.disconnect();
-          ros::Duration(1.0).sleep(); // sleep for one second each time
-          pg_.connect();
-          pg_.start();
-        } catch(std::runtime_error& e2){
-          NODELET_ERROR("%s", e2.what());
-        }*/
+
+        // Try disconnecting from the camera
```

Comment:
@mikepurvis Should I create helper methods to run each of these blocks?
They are also used in `connectCb` and other places.

I'm going to do it, but maybe it should be on another PR, so this one is easier to review.